### DISCUSSION TITLE
How can the inline "cmd+I to continue with copilot" suggestion in VS Code be disabled?

### DISCUSSION THREAD

**Discussion by ege-erdogan** [community/community#147492] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

This is both a question and feedback. 

![image](https://github.com/user-attachments/assets/277611fa-fe59-4409-9de7-1a83bab5e31e)

This inline suggestion pops up when typing things even if copilot is disabled for that file type. So far the only way I've found to make this go away is by completely uninstalling copilot which I don't want to do. How can this be disabled? Nothing in the copilot settings appears to help. 

More as a feedback, it is extremely annoying to have this "suggestion" pop up each time you type something in a code editor. A code editor shouldn't be a place for nudging you to use a certain feature. 

Thanks!

**Comment by likanzhan** [comment]
I have the similar question.

**Comment by dillonkrug** [comment, marked as answer]
+1 this "feature" is an annoyance at best.

EDIT: Right clicking on the message shows a dropdown with "Disable inline chat hint"

**Reply by ericholscher** [reply]
Thanks! This was driving me crazy.

**Reply by crujedo-qualia** [reply]
Right clicking this actually doesn't work in Linux, at least for me. :(

**Reply by heofthetea** [reply]
I too am on linux, and apparently you really cannot right-click that message there. @dillonkrug if you're on vscode, could you maybe check whether disabling the "feature" via right-click is reflected in an entry in your setting.json?
EDIT: found the solution myself, see my reply below

**Reply by DevinCLane** [reply]
thank you

**Reply by diego898** [reply]
My goodness - thank you for that! I would have never guessed to right-click the hint - cant believe it's a feature to just ... permanently display that on all lines - forever?

**Reply by toph-allen** [reply]
Yeah, wow. So glad I found this, it was making typing anything in VS Code really painful.

**Comment by nhonka-eq** [comment]
Agreed is a dumb "feature." Should be reverted.

**Comment by heofthetea** [comment]
Right-clicking doesn't actually work on linux. After conducting some search through the settings, there are two configurations regarding the hints for inline chats. Setting those to false in my `settings.json` fixed the issue for me:
```json
"accessibility.verbosity.inlineChat": false,
 "inlineChat.lineNaturalLanguageHint": false,
```

**EDIT**: I just noticed the first of these options is not actually needed to stop this behaviour. The configuration
```json
"inlineChat.lineNaturalLanguageHint": false
```
on its own suffices.

**Reply by Firgrep** [reply]
Thank you! I'm on linux too and this was annoying as hell.

**Reply by jamesonsaunders** [reply]
Praise be

**Reply by peshal-jha-work** [reply]
Worked on macos, Cheers!

**Comment by levmalcev** [comment]
Other people have already answered you but I would like to thank you for your question.
This feature is SO annoying; what a poor UX choice!

**Comment by molejnik-bct** [comment]
I'm on Linux as well. Interestingly, you can ask Copilot (in vscode's chat) how to disable it and it will tell you how to do it, even provides an interactive button which (when clicked) opens  the settings where it is possible to disable the feature.
![image](https://github.com/user-attachments/assets/978011cd-da34-4d90-8ba1-0f653eeb5ade)
